runtime/src/bank/bank_hash_details.rs

```diff
@@ -162,6 +170,7 @@ impl TryFrom<&Bank> for SlotDetails {
             accounts_delta_hash,
             bank.signature_count(),
             bank.last_blockhash(),
+            bank.wait_get_epoch_accounts_hash(),
```

Comment:
> I think we can query the EAH directly without the "wait" part via this function:

Is the justification here that we must have already frozen the `Bank`, and in doing so, the EAH was ready (or we already waited for it to be ready) ?

>  think there is a case to be made for this refactor and putting should_include_epoch_accounts_hash() inside the wait_ fn, but I think we can handle that in a separate PR (or prior PR).

Cool, I think we agree on putting the `should_include_*()` inside the fetches. Let's land that first

I'm leaning towards keeping it a method on `Bank`. I plan on adjusting this function as well
https://github.com/anza-xyz/agave/blob/9d4867e24b8ac4e7ced5e3122ce844a5b56bcd65/runtime/src/bank.rs#L6398-L6405

Reply:
> do the long call through bank to accounts-db to epoch_accounts_hash_manager and then call try_get_eah() directly.

Finally got back to this PR, an item for consideration:
- With https://github.com/anza-xyz/agave/pull/2477 merged, `Bank::wait_get_epoch_accounts_hash()` now has the check for `self.should_include_epoch_accounts_hash()` internally. Part of the motivation for moving `should` check inside was to avoid duplication elsewhere
- Your comment initially pointed out that this function didn't need the "wait" part, because the EAH should implicitly be ready

These two bullets are a bit in conflict with each other. We could avoid using the "wait" function here; however, we will reintroduce some duplicate code to do so.

One alternative I could see is attaching a comment here in `bank_hash_details.rs` like:
> If the EAH was included in this Bank, the EAH was either already ready or waited on in Bank::freeze(). Either way, the EAH is already ready here so this function will not wait